frontend/src/services/web3-provider/metamask-provider.ts

```diff
@@ -15,6 +16,12 @@ export async function createMetaMaskProvider(): Promise<MetaMaskProvider | undef
   return undefined;
 }
 
+export async function onboardMetaMask() {
```

Comment:
For me it takes like 3 seconds until the download page opens when clicking the metamask button and it is not installed. Would it be possible to show a spinner to indicate that something is happening?

Reply:
Those 3 seconds or so are related to how their library is working internally. 
Check this diagram out: https://docs.metamask.io/guide/onboarding-library.html#onboarding-diagram

I think we cannot improve this much.